### PR TITLE
Javascript/javascript3/week3

### PR DESCRIPTION
**Pull request by qamarfj**

### CODE REVIEW

File: javascript/javascript3/week3/make-art/circle-art.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  draw() {
+    context.fillStyle = this.fillColor;
+    context.beginPath();
+    context.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
+    context.closePath();
+    context.fill();
+  }
+}
+// ranodm color genrator
+function getRandomColor() {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/make-art/circle-art.js:26]: nice function naming

File: javascript/javascript3/week3/make-art/circle-art.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const letters = "0123456789ABCDEF";
+  let color = "#";
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/make-art/circle-art.js]: it's usually better avoid long expressions inside square brackets []
```
const randomLetter = Math.floor(Math.random() * 16)
color+=letters[randomLetter]
```

File: javascript/javascript3/week3/shopping/shopping.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+const mainUL = document.querySelector("#container");
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+  async convertToCurrency(currency) {
+    let isCurrencyFound = false;
+    let price;
+    await fetch("https://api.exchangeratesapi.io/latest?base=DKK")
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping/shopping.js]: don't mix await and then, it's confusing. Best practice is to use async / await, and avoid .then(). 
Rewrite this code in the async / await way and you'll see how much easier it is to read.

File: javascript/javascript3/week3/shopping/shopping.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    totalpriceLabelLi.textContent = "Total Price: ";
+    totalPriceLi.textContent = this.getTotal();
+    ul.appendChild(totalpriceLabelLi);
+    ul.appendChild(totalPriceLi);
+    const li = document.createElement("li");
+    li.appendChild(ul);
+    mainUL.appendChild(li);
+  }
+
+  getUser() {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping/shopping.js]: better to use async / await everywhere.

File: javascript/javascript3/week3/shopping/shopping.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+//Part 3 Assuming dkk as default currency
+const plant = new Product("plant", 50);
+const currencyName = "GBP";
+plant
+  .convertToCurrency(currencyName)
+  .then((price) =>
+    console.log(`${plant.name}'s price in ${currencyName} is : ${price}`)
+  );
+const wrongCurrencyName = "Wrong Currency";
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping/shopping.js]: this code `const wrontCurrencyName ...` will execute _before_ .then() above,
convert to async / await.

**Review by aleksd-dfds — APPROVED** [review]
Good job, a bit of confusion on the .then() chaining. Async / await is your friend, also on a real job.

File: javascript/javascript3/week3/shopping/shopping.js (hunk 5 above)
Comment (qamarfj) [reply javascript/javascript3/week3/shopping/shopping.js]: Thank you for your precious time for the review and valuable comments. I have implemented all the suggested changes.

**Pull request merged**